Quests/QuestNPC.cs

```diff
@@ -81,5 +135,85 @@ public override void PostDraw(NPC npc, SpriteBatch spriteBatch, Color drawColor)
 
             base.PostDraw(npc, spriteBatch, drawColor);
         }
+
+
+        public override void NPCLoot(NPC npc)
+        {
+            // Like three for loops just to check if the NPC killed can count towards the player's quest, not sure if more optimal way?
+            foreach (Player player in Main.player)
```

Comment:
We can probably clean this up a bit, but the logic isn't really that bad. It's only on kill, and we iterate over `modPlayer.CurrentQuests` which isn't going to be that large.